app/src/main/java/ru/valentingordienko/loftcoin/screens/rate/CurrencyDialog.java

```diff
@@ -0,0 +1,72 @@
+package ru.valentingordienko.loftcoin.screens.rate;
+
+import android.app.Dialog;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
+import androidx.fragment.app.DialogFragment;
+
+import ru.valentingordienko.loftcoin.R;
+import ru.valentingordienko.loftcoin.utils.Fiat;
+
+public class CurrencyDialog extends DialogFragment {
+
+    public static final String TAG = "CurrencyDialog";
+
+    private CurrencyDialogListener listener;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+        View view = LayoutInflater.from(requireContext()).inflate(R.layout.dialog_currency, null);
+
+
+        AlertDialog dialog = new AlertDialog.Builder(requireContext())
+                .setView(view)
+                .create();
+
+        View usd = view.findViewById(R.id.usd);
+        View eur = view.findViewById(R.id.eur);
+        View rub = view.findViewById(R.id.rub);
+
+        usd.setOnClickListener(v -> {
```

Comment:
why not use 1 listener instead of 3?

Reply:
Becouse i don't know how i can pass differebt Fiat argument into listener for different situations.

Reply:
good example https://startandroid.ru/ru/uroki/vse-uroki-spiskom/17-urok-10-optimiziruem-realizatsiju-obrabotchikov.html